commission/models/commission_mixin.py

```diff
@@ -37,13 +37,15 @@ def _prepare_agent_vals(self, agent):
 
     def _prepare_agents_vals_partner(self, partner, settlement_type=None):
         """Utility method for getting agents creation dictionary of a partner."""
-        agents = partner.agent_ids
-        if settlement_type:
-            agents = agents.filtered(
-                lambda x: not x.commission_id.settlement_type
-                or x.commission_id.settlement_type == settlement_type
+        return [
+            (0, 0, self._prepare_agent_vals(agent))
+            for agent in partner.agent_ids
+            if settlement_type
+            and (
+                not agent.commission_id.settlement_type
+                or agent.commission_id.settlement_type == settlement_type
```

Comment:
This is not expressing the same condition as before, and I think this is less readable.

Reply:
The idea behind this change was also related to time complexity like my other comment, but I agree that it is harder to read and since this method seems to only be used for single lines, I will change it back